app/controllers/settings/from_address_controller.rb

```diff
@@ -1,14 +1,22 @@
 class Settings::FromAddressController < FormController
   before_action :assign_from_address
 
-  def index; end
+  def index
+    @from_address_presenter = FromAddressPresenter.new(@from_address)
+  end
 
   def create
-    @from_address.update!(from_address_params)
+    from_address_creation = FromAddressCreation.new(
+      from_address: @from_address,
+      from_address_params: from_address_params,
+      email_service: email_service
+    )
 
-    redirect_to settings_from_address_index_path(params[:id])
-  rescue ActiveRecord::RecordInvalid
-    render :index
+    if from_address_creation.save
+      redirect_to settings_from_address_index_path(params[:id])
```

Comment:
if we are going to use the service_id from the service object a la line 30 then we should be consistent in our controller and not use params[:id]